strongbox-storage/strongbox-storage-api/src/test/java/org/carlspring/strongbox/storage/repository/remote/heartbeat/RemoteRepositoryAlivenessCacheManagerTest.java

```diff
@@ -0,0 +1,157 @@
+
+    @BeforeEach
+    void setUp()
+    {
+        remoteRepositoryAlivenessCacheManager = new RemoteRepositoryAlivenessCacheManager(cacheManager);
+    }
+
+    @Test
+    public void isAliveShouldReturnTrueWhenRemoteRepositoryNotCached()
+    {
+        RemoteRepository remoteRepository = createMockedRemoteRepository();
+
+        boolean alive = remoteRepositoryAlivenessCacheManager.isAlive(remoteRepository);
+
+        assertTrue(alive);
+    }
+
+    @Test
+    public void isAliveShouldReturnTrueWhenRemoteRepositoryCachedValueIsTrue()
+    {
+        RemoteRepository remoteRepository = createMockedRemoteRepository();
+
+        initializeCache(true);
+
+        boolean alive = remoteRepositoryAlivenessCacheManager.isAlive(remoteRepository);
+
+        assertTrue(alive);
+    }
+
+    @Test
+    public void isAliveShouldReturnFalseWhenRemoteRepositoryCachedValueIsFalse()
+    {
+        RemoteRepository remoteRepository = createMockedRemoteRepository();
+
+        initializeCache(false);
+        remoteRepositoryAlivenessCacheManager.put(remoteRepository, false);
+
+        boolean alive = remoteRepositoryAlivenessCacheManager.isAlive(remoteRepository);
+
+        assertFalse(alive);
+    }
+
+    @Test
+    public void isAliveShouldReturnTrueWhenRemoteRepositoryCachedValueIsExpired()
+    {
+        // TODO: Figure out how to forcibly expire a cached item.
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideCacheValues")
+    public void putShouldChangeCachedValueImmediately(Boolean initialCacheValue,
+                                                      boolean newCacheValue)
+    {
+        RemoteRepository remoteRepository = createMockedRemoteRepository();
+
+        initializeCache(initialCacheValue);
+
+        remoteRepositoryAlivenessCacheManager.put(remoteRepository, newCacheValue);
+        boolean alive = remoteRepositoryAlivenessCacheManager.isAlive(remoteRepository);
+
+        assertEquals(newCacheValue, alive);
+    }
+
+    private static Stream<Arguments> provideCacheValues()
+    {
+        return Stream.of(
+                Arguments.of(null, false),
+                Arguments.of(null, true),
+                Arguments.of(false, false),
+                Arguments.of(true, true)
+        );
+    }
+
+    private RemoteRepository createMockedRemoteRepository()
+    {
+        RemoteRepository remoteRepository = Mockito.mock(RemoteRepository.class);
+        Mockito.when(remoteRepository.getUrl()).thenReturn(REMOTE_REPOSITORY_URL);
+
+        return remoteRepository;
+    }
+
+    private void initializeCache(Boolean initialValue)
+    {
+        Cache cache = cacheManager.getCache(CacheName.Repository.REMOTE_REPOSITORY_ALIVENESS);
+        Objects.requireNonNull(cache, "remoteRepositoryAliveness cache configuration was not provided");
+
+        cache.put(REMOTE_REPOSITORY_URL, initialValue);
+    }
+
+    @Profile("RemoteRepositoryAlivenessCacheManagerTestConfig")
+    @Import({ HazelcastConfiguration.class,
```

Comment:
@mwvdev probably you only need `HazelcastConfiguration` here, below configurations seems excess.

Reply:
you can create just `CahceManager` like below:
```
    @Bean
    public CacheManager cacheManager(HazelcastInstance hazelcastInstance)
    {
        return new HazelcastCacheManager(hazelcastInstance);
    }
```

Reply:
Great suggestion, I'll use this approach instead.